src/js/common/components/RequiredTermsAcceptanceView.jsx

```diff
@@ -0,0 +1,56 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { checkAcceptance } from '../../user-profile/actions';
+
+import AcceptTermsPrompt from './AcceptTermsPrompt';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+
+class RequiredTermsAcceptanceView extends React.Component {
+  componentWillMount() {
+    // I get errors and the app crashes if this line is called.
+    // this.props.checkAcceptance();
```

Comment:
@bshyong I think the crash may be fixed. It still bails when getting the 404, and I think that's the appropriate response from the backend, so we'll need to handle that on the client. You're right that the 404 does indicate that the user hasn't accepted the terms.